File: app/src/main/java/com/vn/ezlearn/activity/MyApplication.java
package com.vn.ezlearn.activity;

import android.app.Application;
import android.content.Context;
import android.util.Log;

import com.vn.ezlearn.config.EzlearnService;
import com.vn.ezlearn.modelresult.CategoryResult;

import rx.Scheduler;

/**
 * Created by manhi on 4/1/2017.
 */

public class MyApplication extends Application {
    private static final String TAG = MyApplication.class.getSimpleName();
    private EzlearnService ezlearnService;
    private Scheduler defaultSubscribeScheduler;
    public int REALM_VERSION = 1;
    private CategoryResult categoryResult;

    @Override
    public void onCreate() {
        super.onCreate();
        Log.d(TAG, "Subscribed to news topic");
    }

    public EzlearnService getEzlearnService() {
        if (ezlearnService == null) {
            ezlearnService = EzlearnService.Factory.create(this);
        }
        return ezlearnService;
    }

    public static MyApplication with(Context context) {
        return (MyApplication) context.getApplicationContext();
    }

    public CategoryResult getCategoryResult() {
        return categoryResult;
    }

    public void setCategoryResult(CategoryResult categoryResult) {
        this.categoryResult = categoryResult;
    }
}

File: app/src/main/java/com/vn/ezlearn/activity/MyApplication.kt
package com.vn.ezlearn.activity

import android.app.Application
import android.content.Context
import android.graphics.Typeface
import android.os.Build
import com.vn.ezlearn.config.EzlearnService
import com.vn.ezlearn.modelresult.CategoryResult







/**
 * Created by manhi on 4/1/2017.
 */

class MyApplication : Application() {
    var categoryResult: CategoryResult? = null

    fun getEzlearnService(): EzlearnService = EzlearnService.Factory.create(this)

    companion object {
        fun with(context: Context): MyApplication = context.applicationContext as MyApplication
    }

    override fun onCreate() {
        super.onCreate()
        changeAppFont()
    }

    private fun changeAppFont() {
        setDefaultFont(this, "DEFAULT", "normal.ttf")
        setDefaultFont(this, "DEFAULT_BOLD", "bold.ttf")
        setDefaultFont(this, "DEFAULT_ITALIC", "italic.ttf")
//
//        setDefaultFont(this, "MONOSPACE", "fonts/FuturaLT-Oblique.ttf");
//        setDefaultFont(this, "SANS_SERIF", "fonts/FuturaLT.ttf");
        setDefaultFont(this, "SERIF", "normal.ttf")
    }

    private fun setDefaultFont(context: Context,
                               staticTypefaceFieldName: String, fontAssetName: String) {
        val regular = Typeface.createFromAsset(context.assets,
                fontAssetName)
        replaceFont(staticTypefaceFieldName, regular)
    }

    private fun replaceFont(staticTypefaceFieldName: String,
                            newTypeface: Typeface) {
        if (isVersionGreaterOrEqualToLollipop()) {
            val newMap = HashMap<String, Typeface>()
            newMap.put("sans-serif", newTypeface)
            try {
                val staticField = Typeface::class.java
                        .getDeclaredField("sSystemFontMap")
                staticField.isAccessible = true
                staticField.set(null, newMap)
            } catch (e: NoSuchFieldException) {
                e.printStackTrace()
            } catch (e: IllegalAccessException) {
                e.printStackTrace()
            }

        } else {
            try {
                val staticField = Typeface::class.java
                        .getDeclaredField(staticTypefaceFieldName)
                staticField.isAccessible = true
                staticField.set(null, newTypeface)
            } catch (e: NoSuchFieldException) {
                e.printStackTrace()
            } catch (e: IllegalAccessException) {
                e.printStackTrace()
            }

        }
    }

    private fun isVersionGreaterOrEqualToLollipop(): Boolean =
            android.os.Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP
}
